pragma-node/src/handlers/entries/subscribe_to_entry.rs

```diff
@@ -0,0 +1,390 @@
+            }
+        } else {
+            let error_msg = "Could not serialize ack message.";
+            send_err_to_socket(socket, error_msg).await;
+        }
+    } else {
+        let error_msg = "Invalid message type. Please check the documentation for more info.";
+        send_err_to_socket(socket, error_msg).await;
+    }
+}
+
+/// Send the current median entries to the client.
+async fn send_median_entries(
+    socket: &mut WebSocket,
+    state: &AppState,
+    subscription: &CurrentSubscription,
+) -> Result<(), EntryError> {
+    if subscription.is_empty() {
+        return Ok(());
+    }
+    let response = match get_subscribed_pairs_medians(state, subscription).await {
+        Ok(response) => response,
+        Err(e) => {
+            send_err_to_socket(socket, &e.to_string()).await;
+            return Err(e);
+        }
+    };
+
+    if let Ok(json_response) = serde_json::to_string(&response) {
+        if socket.send(Message::Text(json_response)).await.is_err() {
+            send_err_to_socket(socket, "Could not send prices.").await;
+        }
+    } else {
+        send_err_to_socket(socket, "Could not serialize prices.").await;
+    }
+    Ok(())
+}
+
+/// Get the current median entries for the subscribed pairs and sign them as Pragma.
+async fn get_subscribed_pairs_medians(
+    state: &AppState,
+    subscription: &CurrentSubscription,
+) -> Result<SubscribeToEntryResponse, EntryError> {
+    let median_entries = get_all_median_entries(state, subscription).await?;
+    let mut response: SubscribeToEntryResponse = Default::default();
+    let now = chrono::Utc::now().timestamp();
+    for entry in median_entries {
+        let median_price = entry.median_price.clone();
+        let mut oracle_price: AssetOraclePrice = entry
+            .try_into()
+            .map_err(|_| EntryError::InternalServerError)?;
+
+        let signature = sign_median_price(
+            &state.pragma_signer,
+            &oracle_price.global_asset_id,
+            now as u64,
+            median_price,
+        )?;
+
+        oracle_price.signature = signature;
+        response.oracle_prices.push(oracle_price);
+    }
+    response.timestamp_s = now.to_string();
+    Ok(response)
+}
+
+async fn get_all_median_entries(
+    state: &AppState,
+    subscription: &CurrentSubscription,
+) -> Result<Vec<MedianEntryWithComponents>, EntryError> {
+    let index_pairs = subscription.get_subscribed_spot_pairs();
+    let perp_pairs = subscription.get_subscribed_perp_pairs();
+
+    let (index_median_entries, mark_median_entries_result) = tokio::join!(
+        compute_index_median_entries(state, index_pairs),
+        compute_mark_median_entries(state, perp_pairs)
+    );
+
+    let mut median_entries = vec![];
+    median_entries.extend(index_median_entries?);
+    median_entries.extend(mark_median_entries_result?);
+    Ok(median_entries)
+}
+
+async fn compute_index_median_entries(
```

Comment:
I think we can make this whole code cleaner by leveraging a `Price` trait and having two structs `IndexPrice` and `MarkPrice` that implement it